rs/sns/testing/src/sns.rs

```diff
@@ -1,27 +1,49 @@
 use candid::{CandidType, Encode};
 use canister_test::Wasm;
-use ic_base_types::CanisterId;
-use ic_management_canister_types::CanisterStatusType as CanisterStatusResultStatus;
+use ic_base_types::{CanisterId, PrincipalId};
 use ic_management_canister_types_private::CanisterInstallMode;
-use ic_nervous_system_agent::sns::Sns;
+use ic_nervous_system_agent::{
+    helpers::await_with_timeout,
+    helpers::nns::propose_to_deploy_sns_and_wait,
+    helpers::sns::{
+        await_swap_lifecycle, get_caller_neuron, participate_in_swap, propose_and_wait,
+    },
+    nns::ledger::transfer,
+    pocketic_impl::PocketIcAgent,
+    sns::{swap::SwapCanister, Sns},
+    CallCanisters, ProgressNetwork,
+};
+use ic_nervous_system_clients::canister_status::CanisterStatusType;
 use ic_nervous_system_integration_tests::{
     create_service_nervous_system_builder::CreateServiceNervousSystemBuilder,
-    pocket_ic_helpers::nns::governance::propose_to_deploy_sns_and_wait,
-    pocket_ic_helpers::sns::{
-        governance::{
-            propose_to_upgrade_sns_controlled_canister_and_wait,
-            EXPECTED_UPGRADE_DURATION_MAX_SECONDS,
+    pocket_ic_helpers::{
+        install_canister_on_subnet,
+        nns::ledger::mint_icp,
+        sns::{
+            governance::EXPECTED_UPGRADE_DURATION_MAX_SECONDS,
+            swap::{is_auto_finalization_status_committed_or_err, swap_direct_participations},
         },
-        swap::{await_swap_lifecycle, smoke_test_participate_and_finalize},
     },
-    pocket_ic_helpers::{await_with_timeout, install_canister_on_subnet},
 };
-use ic_nns_common::pb::v1::ProposalId;
+use ic_nns_common::pb::v1::NeuronId;
 use ic_nns_constants::ROOT_CANISTER_ID;
-use ic_sns_governance_api::pb::v1::UpgradeSnsControlledCanister;
+use ic_nns_governance_api::pb::v1::create_service_nervous_system::{
+    initial_token_distribution::developer_distribution::NeuronDistribution, SwapParameters,
+};
+
+use ic_nns_test_utils::common::modify_wasm_bytes;
+use ic_sns_governance_api::pb::v1::{
+    manage_neuron::Follow, proposal::Action, Proposal, UpgradeSnsControlledCanister,
+};
 use ic_sns_swap::pb::v1::Lifecycle;
+use icp_ledger::{AccountIdentifier, Memo, Tokens, TransferArgs, DEFAULT_TRANSFER_FEE};
 use pocket_ic::nonblocking::PocketIc;
 
+use crate::utils::NNS_NEURON_ID;
+
+// As per https://internetcomputer.org/docs/current/developer-docs/defi/rosetta/icp_rosetta/construction_api/staking/#dissolve-delay
+const MINIMUM_DISSOLVE_DELAY_TO_VOTE: u32 = 15778800;
+
 // TODO @rvem: I don't like the fact that this struct definition is copy-pasted from 'canister/canister.rs'.
 // We should extract it into a separate crate and reuse in both canister and this crates.
```

Comment:
Note that the source of truth for a canister's API should be the Candid file, not some Rust type. So in my view, it's fine to duplicate the Rust types (assume that in an ideal world they would be generated)